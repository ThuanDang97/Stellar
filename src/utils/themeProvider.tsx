import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
