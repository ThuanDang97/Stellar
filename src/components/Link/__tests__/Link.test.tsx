import { screen } from '@testing-library/react'

// Components
import LinkComponent from '@components/Link'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

describe('Link Component', () => {
  const component = <LinkComponent text="Link" href="./abc" />

  test('Component Link matchers DOM Snapshot', () => {
    const { container } = renderWithTheme(component)

    expect(container).toMatchSnapshot()
  })

  test('Component Link should render default correctly', () => {
    renderWithTheme(component)

    expect(screen.getByTestId('link')).toHaveTextContent('Link')
  })
})
