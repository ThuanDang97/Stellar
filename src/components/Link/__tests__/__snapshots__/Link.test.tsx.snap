// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Component Component Link matchers DOM Snapshot 1`] = `
<div>
  <a
    class="LinkStyled__ButtonLinkStyled-sc-626068-1 grtQrl"
    data-testid="link"
    href="/abc"
  >
    Link
  </a>
</div>
`;
